### ISSUE TITLE
nvic.h should not include WEAK in prototypes

### ISSUE THREAD

**Issue by jimparis** [libopencm3/libopencm3#568]
None of the `nvic.h` files (`cm3/nvic.h` or the generated ones) should contain WEAK in the prototypes.  Otherwise, even the "real" definitions end up as weak symbols, and it's random chance as to whether your final binary gets the libopencm3 weak one, or the real one that you defined in your code.  For example, after a default build of libopencm3-examples:
```shell
$ arm-none-eabi-nm examples/stm32/f4/stm32f4-discovery/tick_blink/tick_blink.o | grep sys_
00000000 W sys_tick_handler
$ arm-none-eabi-nm libopencm3/lib/libopencm3_stm32f4.a | grep sys_
00000000 W sys_tick_handler
```
Both weak!  The linker happens to picks the right one by chance in the examples, but I have another project where it pulls in the wrong one.  After removing WEAK from the prototypes, everything works correctly:

```shell
$ arm-none-eabi-nm examples/stm32/f4/stm32f4-discovery/tick_blink/tick_blink.o | grep sys_
00000000 T sys_tick_handler
$ arm-none-eabi-nm libopencm3/lib/libopencm3_stm32f4.a | grep sys_
00000000 W sys_tick_handler
```

**Comment by ChuckM** [comment]
:+1:

**Comment by karlp** [comment]
Closed via merge of #569
